# HG changeset patch
# User E. Joshua Rigler <relgire@gmail.com>
# Date 1251790299 -7200
#      Tue Sep 01 09:31:39 2009 +0200
# Branch release-3-2-x
# Node ID 48a9c78cfbc94c37ced29ffd667ec06a50754bcb
# Parent  f3e70f64c30c4bb52d8ff3ffb4676477720140ee
datestr: set tm.isdst to -1 before calling mktime

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-08-26  E. Joshua Rigler  <relgire@gmail.com>
+
+	* time/datestr.m: Set tm.isdst to -1 before calling mktime.
+
 2009-08-06  John W. Eaton  <jwe@octave.org>
 
 	* statistics/base/std.m: Correctly work along singleton dimension.
 	From Christoph Ellenberger <C.Ellenberger@gmx.net>.
 
 2009-08-06  Olaf Till <olaf.till@uni-jena.de>
 
 	* geometry/griddata.m: Linearize arrays.
diff --git a/scripts/time/datestr.m b/scripts/time/datestr.m
--- a/scripts/time/datestr.m
+++ b/scripts/time/datestr.m
@@ -272,17 +272,19 @@ function retval = datestr (date, f, p)
     tm.year = vi(1) - 1900;
     tm.mon = vi(2) - 1;
     tm.mday = vi(3);
     tm.hour = vi(4);
     tm.min = vi(5);
     sec = vi(6);
     tm.sec = fix (sec);
     tm.usec = fix (rem (sec, 1) * 1e6);
-
+    ## Force mktime to check for DST.
+    tm.isdst = -1
+ 
     str = strftime (df, localtime (mktime (tm)));
 
     if (i == 1)
       retval = str;
     else
       retval = [retval; str];
     endif
 
